crates/spirv-std/src/vector.rs

```diff
@@ -23,3 +25,54 @@ unsafe impl Vector<u32, 4> for glam::UVec4 {}
 unsafe impl Vector<i32, 2> for glam::IVec2 {}
 unsafe impl Vector<i32, 3> for glam::IVec3 {}
 unsafe impl Vector<i32, 4> for glam::IVec4 {}
+
+/// Abstract trait representing a SPIR-V scalar or vector type.
+///
+/// # Safety
+/// Implementing this trait on non-simd-vector/scalar types breaks assumptions of other unsafe code, and
+/// should not be done.
+pub unsafe trait ScalarOrVector {
+    /// The scalar type
+    type Scalar;
+
+    /// The dimension of the vector, or 1 if it is a scalar
+    const DIM: usize;
```

Comment:
Can you use [std](https://doc.rust-lang.org/stable/std/index.html)::[num](https://doc.rust-lang.org/stable/std/num/index.html)::[NonZeroUsize](https://doc.rust-lang.org/stable/std/num/struct.NonZeroUsize.html#) here?

Reply:
Yes, theoretically you could do it, but I am not sure if it is currently supported in Rust-Gpu.